airflow/providers/snowflake/transfers/s3_to_snowflake.py

```diff
@@ -58,22 +77,38 @@ def __init__(
         file_format: str,
         schema: str,  # TODO: shouldn't be required, rely on session/user defaults
         columns_array: Optional[list] = None,
+        warehouse: Optional[str] = None,
         autocommit: bool = True,
         snowflake_conn_id: str = 'snowflake_default',
+        role: Optional[str] = None,
+        authenticator: Optional[str] = None,
+        session_parameters: Optional[dict] = None,
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
         self.s3_keys = s3_keys
         self.table = table
+        self.warehouse = warehouse
         self.stage = stage
         self.file_format = file_format
         self.schema = schema
         self.columns_array = columns_array
         self.autocommit = autocommit
         self.snowflake_conn_id = snowflake_conn_id
+        self.role = role
+        self.authenticator = authenticator
+        self.session_parameters = session_parameters
 
     def execute(self, context: Any) -> None:
-        snowflake_hook = SnowflakeHook(snowflake_conn_id=self.snowflake_conn_id)
+        snowflake_hook = SnowflakeHook(
+            snowflake_conn_id=self.snowflake_conn_id,
+            warehouse=self.warehouse,
+            database=self.database,
```

Comment:
Where do you set the value of `self.database` ?

Reply:
My mistake. I think in this case is not necessary, so I'll delete it from here

Reply:
Why? like schema you can also select different DBs.
Do you see value in limiting it only to the DB defined in the connection?